geode-core/src/main/java/org/apache/geode/cache/client/internal/EndpointManagerImpl.java

```diff
@@ -56,17 +56,17 @@ public EndpointManagerImpl(String poolName, DistributedSystem ds, CancelCriterio
 
   @Override
   public Endpoint referenceEndpoint(ServerLocation server, DistributedMember memberId) {
-    Endpoint endpoint = endpointMap.get(server);
+    Endpoint endpoint = endpointMap.get(memberId);
```

Comment:
This is what is causing the ConnectionPoolImplJUnitTest failure.  That test creates two servers and a pool in the same JVM, so both servers have the same membership ID.  If you delete that test and replace it with a distributedTest that creates the servers in different JVMs it ought to pass.

Reply:
Actually I think this is a problem that needs to be brought to the dev email list.  Either the PR needs to change or we need to disallow creating multiple CacheServers in the same cache.